### PR TITLE
docs: add design doc for cardinality estimation module evaluation

### PR DESCRIPTION
**Pull request by qw4990**
### What problem does this PR solve?

Issue Number: close #xxx

Problem Summary:

### What is changed and how it works?

docs: add design doc for cardinality estimation module evaluation

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: docs/design/2021-11-16-CE-evaluation.md [hunk 1]
```diff
@@ -0,0 +1,102 @@
+# Proposal: Support Analyze Predicate Columns
```
Comment (Reminiscent) [inline docs/design/2021-11-16-CE-evaluation.md:1]: The title should be changed.

Comment (qw4990) [reply docs/design/2021-11-16-CE-evaluation.md]: updated...

File: docs/design/2021-11-16-CE-evaluation.md [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+```
+> trace target='estimation' select * from t where a < 10 and b > 10;
++------------+ 
+| Dump_link  | 
++------------+ 
+| xxx.zip    | 
++------------+
+```
+
+All trace information(intermediate results) is stored in a JSON file:
```
Comment (winoros) [inline docs/design/2021-11-16-CE-evaluation.md:69]: How we get the error rate via these json output?

Comment (qw4990) [reply docs/design/2021-11-16-CE-evaluation.md:69]: This JSON file just contains the estimated rows by the optimizer, and we have to use the external tool to get these q-errors.

**Pull request closed without merging**